WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -410,11 +410,7 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
             showNoSites()
             return
         }
-
-        showSitePicker(for: mainBlog)
-        showBlogDetails(for: mainBlog)
-        updateNavigationTitle(for: mainBlog)
-        updateSegmentedControl(for: mainBlog, switchTabsIfNeeded: true)
+        self.blog = mainBlog
```

Comment:
> This seems like the right direction towards a more DRY approach.

Yes, It's easy to maintain if we toward DRY in code. 

> - Do you foresee any issues with the extra calls to createFABIfNeeded() and fetchPrompt(for: newBlog) when this is set?

About the extra call, it will have minor effect on the `createFABIfNeeded()`:
- Inside the `createFABIfNeeded` it increase the counting for the `CreateButtonCoordinator.shownTooltipCount`.
If we has extra cell on it, it will have extra counting for the tool tips.

Because it have the side effect above if we only tweak code here, 
It will be better to refactor for DRY all at once in the future, so I revert the commit for this.